ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java

```diff
@@ -58,7 +60,18 @@
 public class BackupCheckerThread extends ReplicaThread {
 
   private final Logger logger = LoggerFactory.getLogger(BackupCheckerThread.class);
+  protected final BackupCheckerFileManager fileManager;
   public static final String DR_Verifier_Keyword = "dr";
+  public static final String MISSING_KEYS_FILE = "missingKeys";
+  public static final String REPLICA_STATUS_FILE = "replicaCheckStatus";
+  /**
+   * To create a file, you need both CREATE + WRITE/APPEND and not just CREATE or just WRITE/APPEND.
+   * For TRUNCATE_EXISTING to work, the file must exist and CREATE does not seem to have any effect there.
+   */
+  public static final EnumSet<StandardOpenOption> CREATE_TRUNC_WRITE = EnumSet.of(StandardOpenOption.CREATE,
```

Comment:
Personally I like to encapsulate all the file system API options to the file manager. 
The caller here only cares it's an append or fresh write. 
File manager decides the API options. File manager knows if the file exists or not. If it doesn't exist, it create it. If it exists, depends on it's an append or fresh write, it may truncate or don't truncate.

Reply:
WRITE = just open the file for write, don't create, don't trunc, fail if file absent
WRITE + TRUNC =  just open the file for write, don't create, trunc, fail if file absent
Add CREATE to above means create the file if absent. 
I could make the code in this class thinner by excluding some options and adding more logic in the file mgr using if-else stmts, but i am trying to not to be too specific for use case. And using a set of options and leveraging the Files package to create a file is much easier. No if-else statements. In the end, the result is the same with lesser if-else statements.

Reply:
I don't know in which case the caller wants the file write to fail if the file doesn't absent. 
In the health checker, from caller's point of view, it only wants two things,
1. WRITE at the beginning = write from the beginning, if file exists, then truncate. If not, then create.
2. APPEND = if the file exists, continue to write. If it doesn't, create the file and write.

Different Java API and library may require different parameter options. Expose the option to the user/caller may limit the choice of the file manager's implementation.

But I leave it to you.

Reply:
removed the options to allow for flexible filemgr impl